src/wp-includes/formatting.php

```diff
@@ -579,6 +588,13 @@ function wpautop( $pee, $br = true ) {
 
 	// Optionally insert line breaks.
 	if ( $br ) {
+
+		// Multi-line HTML comments. (Opening)
+		$pee = preg_replace_callback( '/<\!\-\-.*?\n/s', '_autop_newline_preservation_helper', $pee );
```

Comment:
IIRC, `-` outside of character classes `[]` don't need to be escaped, nor does the `!` need to be escaped in regex as it doesn't have any special meaning other than at the start of an assertion `(?!...)`.

Escaping them anyway may be interpreted by the regex engine as looking for a literal `\` character.

Reply:
hey @jrfnl thanks for the feedback, good point! I'll remove the excessive escaping, regulex agrees they are not needed - https://jex.im/regulex/#!flags=&re=%2F%3C!--.*%3F%5Cn%2Fs

Reply:
@adamsilverstein Please be careful using those kind of online tools. They may not use the same regex engine as where you are using the regex - as is the case here.
The Regulex tool is javascript based and JS uses a different regex engine as PHP.
For the future, for PHP PCRE regexes, you may want to consider using this tool: https://regex101.com/
Also note that the PHP Mbstring extension uses yet another regex engine .... ;-)

Reply:
OOOhh, excellent - thanks for the tip! I also use `patterns` on my mac which has language selection so hopefully better. Removed the un-needed escaping in https://github.com/WordPress/wordpress-develop/pull/72/commits/c75cfdb44d12de49b534a312faba9af270ee22ed